src/librustc/ty/mod.rs

```diff
@@ -1070,48 +1100,43 @@ impl<'tcx> Generics {
         false
     }
 
+    /// Returns the `GenericParamDef` with the given index.
+    pub fn param(&'tcx self, index: u32, tcx: TyCtxt<'tcx>) -> &'tcx GenericParamDef {
```

Comment:
That's funny, I added something like this in one of my own PRs (I called it `param_at`).